pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -900,11 +900,21 @@ def init_rNLP(self, config, add_oa_cuts=True):
             )
             # Add OA cut
             if add_oa_cuts:
-                dual_values = (
-                    list(m.dual[c] for c in MindtPy.constraint_list)
-                    if config.calculate_dual_at_solution
-                    else None
-                )
+                if (
+                    self.config.nlp_solver == 'cyipopt'
+                    and self.objective_sense == minimize
+                ):
+                    dual_values = (
+                        list(-1 * m.dual[c] for c in MindtPy.constraint_list)
+                        if config.calculate_dual_at_solution
+                        else None
+                    )
+                else:
+                    dual_values = (
+                        list(m.dual[c] for c in MindtPy.constraint_list)
+                        if config.calculate_dual_at_solution
+                        else None
+                    )
```

Comment:
This code is there to support the dual issues with cyipopt in #2831. This works for the current implementation, but it will break if the issue is solved, although it might require a larger effort to unify the duals for NLP solvers in Pyomo.